Sources/Commands/SwiftTool.swift

```diff
@@ -109,3 +123,21 @@ public class SwiftTool<Mode: Argument, OptionType: Options> {
         }
     }
 }
+
+/// Returns path of the nearest directory containing the manifest file w.r.t
+/// current working directory.
+private func findPackageRoot() -> AbsolutePath? {
```

Comment:
This isn't something to spend time generalizing now, but I would love to at some point have a method like this that takes a path and a predicate block and returns the lowest path for which the predicate is true.  It isn't used super-often, but when it is it would be useful to hide away the implementation.